### PR TITLE
Message Header fix + Add connection pool singleton for NATS/JS connection

### PR DESCRIPTION
**Pull request by Zetanova**
There is an issue with the direct output of the nats msg instance to n8n.
The header fields need to be manually copied or a "output is large" warning be shown 
and the workflow is not usable

### CODE REVIEW

**Comment by Zetanova** [comment]
Sry, for the many commits, my skills with git rebase are very limited.

How to exclude the package.json from a future PR?

File: nodes/Nats.service.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+export class NatsConnectionHandle implements Disposable {
+
+	constructor(private service: NatsService, public connection: NatsConnection, private token: object) {
+	}
+
+	[Symbol.dispose]() {
+		if (this.service) {
+			//todo how can this.service be undefined
```
Comment (Hugome) [inline nodes/Nats.service.ts:110]: Does a
```js
delete this.service
```
Could do the trick ?

Comment (Zetanova) [reply nodes/Nats.service.ts:110]: This is the new explicit resource pattern in typescript 5.2.

But I didn't add the last 3 commits to the PR. 
Github make something automatic.

After the header fix, I tried to make a nats connection as singleton instance for each credential.
Only working on n8n execution workflow layer right now.

**Comment by Hugome** [comment]
Thanks a lot for all the contributions, I'm gonna try a bit a new build and release a version in the next days :)
(Don't worry about the package/lock problem, I'm gonna clean it on a side branch before final merging :+1:)
There is some changes i want to do in the documentation/README before releasing :+1:

**Pull request merged**